text/0314-cdk-construct-hub.md

```diff
@@ -0,0 +1,651 @@
+
+   The Construct Hub will have permissions to read from (and, if applicable, to
+   decrypt) the S3 bucket that is used to hold the object denoted by
+   `assemblyUri`. That bucket may have a Lifecycle configuration that cleans up
+   objects after a few days, as objects need not be retained there after they
+   have been successfully ingested by the Construct Hub (which will own and
+   maintain a copy of this artifact).
+
+   Construct Hub comes with a built-in integration function that listents to
+   changes on the [npmjs.com registry](https://skimdb.npmjs.com/registry)
+   database. Operators of private instances are able to interface with other
+   package registries (including private ones), such as CodeArtifact
+   repositories, etc.
+
+1. **Ingestion:** A Lambda function then picks up messages from the SQS queue
+   and prepares the artifacts consumed by the front-end application, stored in a
+   dedicated S3 bucket using the following key format:
+   `assemblies/${assembly.name}/v${assembly.version}/assembly.json`
+
+   This function validates the contents of the `assembly` to ensure the message
+   was well-formed. In case the message is found to be incoherent, it is sent to
+   a dead-letter queue and an alarm is triggered.
+
+   - Optionally (this information may not be necessary, in particular for the
+     minimal viable product), it also creates or updates the
+     `assemblies/${assembly.name}/versions.json` object, which contains a list of
+     all versions ever indexed for the package, matched to the versions' current
+     status (`deprecated`, `latest`, ...):
+
+     Name        | Description
+     ------------|----------------------------------------------------------------
+     `v${major}` | All versions part of this major version stream, and their status
+
+     [SemVer]: https://semver.org/spec/v2.0.0.html
+
+     In order to ensure consistency of the contents of the `versions.json`
+     object, its content will always be generated from the live inventory of
+     objects in S3 (and not incrementally using the update trigger event), and
+     the last event timestamp will be written in an object metadata attribute.
+     The S3 Bucket will have object versioning enabled, and once a new version
+     of the `versions.json` object has been created, its metadata will be
+     checked against any previous versions of the object, and the "current"
+     version will be set to the one with the latest event timestamp if needed.
+
+   The operator of private instances can provide an optional SNS topic which
+   will be notified once the information for a new package version is ready to
+   be browsed.
+
+1. **Doc-Gen:** A series of Lambda functions prepare language-specific assembly
+   files for each configured language, with adjusted naming conventions, and
+   updated sample code fragments, and proceeds to store those at:
+   `assemblies/${assembly.name}/v${assembly.version}/assembly-${lang}.json`.
```

Comment:
I don't like the `assemblies` term here. From an external perspective. This is just the metadata of a library/module/package in the construct hub.

```suggestion
   `assemblies/${assembly.name}/v${assembly.version}/assembly-${lang}.json`.
```